sbom/sbom_format.go

```diff
@@ -0,0 +1,101 @@
+package sbom
+
+import (
+	"fmt"
+	"mime"
+
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/paketo-buildpacks/packit/v2/sbom/internal/formats/cyclonedx13"
+	"github.com/paketo-buildpacks/packit/v2/sbom/internal/formats/syft2"
+)
+
+// TODO: refactor the version lookup part
+var syftFormats map[string]sbom.FormatID = map[string]sbom.FormatID{
+	"latest": syft.JSONFormatID,
+	"2.0.2":  syft2.ID,
+}
+
+var cyclonedxFormats map[string]sbom.FormatID = map[string]sbom.FormatID{
+	"latest": syft.CycloneDxJSONFormatID,
+	"1.3":    cyclonedx13.ID,
+}
+
+var spdxFormats map[string]sbom.FormatID = map[string]sbom.FormatID{
+	"latest": syft.SPDXJSONFormatID,
+}
+
+var additionalFormats []sbomFormat
+
+func init() {
+	additionalFormats = []sbomFormat{
+		newSBOMFormat(cyclonedx13.Format()),
+		newSBOMFormat(syft2.Format()),
+	}
+}
+
+// An experimental type added to support more SBOM formats
+// It extends the Syft sbom.Format interface
+type sbomFormat struct {
+	sbom.Format
+}
+
+func newSBOMFormat(format sbom.Format) sbomFormat {
+	return sbomFormat{
+		Format: format,
+	}
+}
+
+func (f sbomFormat) Extension() string {
+	switch f.ID() {
+	case syft.CycloneDxJSONFormatID, cyclonedx13.ID:
+		return "cdx.json"
+	case syft.SPDXJSONFormatID:
+		return "spdx.json"
+	case syft.JSONFormatID, syft2.ID:
+		return "syft.json"
+	default:
+		return ""
+	}
+}
+
+func formatByMediaType(mediaType string) (sbomFormat, error) {
+	baseType, params, err := mime.ParseMediaType(mediaType)
+	if err != nil {
+		return sbomFormat{}, fmt.Errorf("failed to parse SBOM media type: %w", err)
+	}
+	// TODO: semver version parsing?
+	version, ok := params["version"]
+	if !ok {
+		version = "latest"
+	}
+	var selected sbom.FormatID
+	switch baseType {
+	case CycloneDXFormat:
+		selected = cyclonedxFormats[version]
+	case SPDXFormat:
+		selected = spdxFormats[version]
+	case SyftFormat:
+		selected = syftFormats[version]
+	default:
+		return sbomFormat{}, fmt.Errorf("unsupported SBOM format: '%s'", mediaType)
+	}
+
+	if selected == sbom.FormatID("") {
+		return sbomFormat{}, fmt.Errorf("version '%s' is not supported for SBOM format '%s'", version, baseType)
+	}
```

Comment:
Why is it necessary to check `selected` against `sbom.FormatID("")`? If there's no matching format/version combo, won't selected just be equal to `""`?

Reply:
Resolved in latest version.